src/phoenix/db/insertion/dataset.py

```diff
@@ -213,6 +325,29 @@ async def add_dataset_examples(
                 f"{dataset_example_id=}"
             )
             raise
+
+        # Collect split assignments for bulk insert later
+        for split_name in example.splits:
+            split_id = split_name_to_id.get(split_name)
+            if split_id is not None:
+                split_assignments.append((dataset_example_id, split_id))
```

Comment:
we can do all the split related operations after the iteration, so we don't need to materialize the whole list upfront
```suggestion
                split_assignments.append((dataset_example_id, name))
```